openapi_metadata.go

```diff
@@ -0,0 +1,423 @@
+	validateTags := strings.Split(validateTag, ",")
+	if ok && slices.Contains(validateTags, "required") {
+		params.SchemaRef.Value.Required = append(params.SchemaRef.Value.Required, params.FieldName)
+	}
+	for _, validateTag := range validateTags {
+		if strings.HasPrefix(validateTag, "min=") {
+			min, err := strconv.Atoi(strings.Split(validateTag, "=")[1])
+			if err != nil {
+				slog.Warn("Min might be incorrect (should be integer)", "error", err)
+			}
+
+			if params.Property.Type.Is(openapi3.TypeInteger) {
+				minPtr := float64(min)
+				params.Property.Min = &minPtr
+			} else if params.Property.Type.Is(openapi3.TypeString) {
+				//nolint:gosec // disable G115
+				params.Property.MinLength = uint64(min)
+			}
+		}
+		if strings.HasPrefix(validateTag, "max=") {
+			max, err := strconv.Atoi(strings.Split(validateTag, "=")[1])
+			if err != nil {
+				slog.Warn("Max might be incorrect (should be integer)", "error", err)
+			}
+			if params.Property.Type.Is(openapi3.TypeInteger) {
+				maxPtr := float64(max)
+				params.Property.Max = &maxPtr
+			} else if params.Property.Type.Is(openapi3.TypeString) {
+				//nolint:gosec // disable G115
+				maxPtr := uint64(max)
+				params.Property.MaxLength = &maxPtr
+			}
+		}
+	}
+}
+
+// DescriptionMetadataParser extracts the "description" tag from a struct field and assigns
+// it to the Description property of the OpenAPI schema.
+func MetadataParserDescription(params MetadataParserParams) {
+	description := params.Field.Tag.Get("description")
+	params.Property.Description = description
+}
+
+// XMLMetadataParser extracts the "xml" tag from a struct field and sets the XML
+// representation of the OpenAPI schema. If the xml tag is "-", the function returns
+// without making any changes. The XML field name is determined from the tag or defaults
+// to the field's name. The function also checks for additional XML attributes such as
+// "attr" and "wrapped" to set the corresponding properties in the OpenAPI XML schema.
+func MetadataParserXML(params MetadataParserParams) {
+	xmlField := params.Field.Tag.Get("xml")
+	if xmlField == "-" || xmlField == "" {
+		return
+	}
+
+	if params.Property == nil {
+		params.Property = &openapi3.Schema{
+			XML: &openapi3.XML{
+				Attribute: false,
+			},
+		}
+	}
+
+	xmlFieldName := strings.Split(xmlField, ",")[0]
+	if xmlFieldName == "-" {
+		return
+	}
+	if xmlFieldName != "" {
+		params.Property.XML = &openapi3.XML{
+			Name: xmlFieldName,
+		}
+	}
+	xmlFields := strings.Split(xmlField, ",")
+	if slices.Contains(xmlFields, "attr") {
+		params.Property.XML.Attribute = true
+	}
+	if slices.Contains(xmlFields, "wrapped") {
+		params.Property.XML.Wrapped = true
+	}
+
+	if params.Field.Type.Kind() == reflect.Struct {
+		if params.Property.Properties == nil {
+			params.Property.Properties = make(map[string]*openapi3.SchemaRef)
+		}
+		processXMLStructChildren(params)
+	} else if params.Field.Type.Kind() == reflect.Slice {
```

Comment:
Assume this would be similar to [`dive`](https://github.com/go-fuego/fuego/blob/main/openapi.go#L292). You're gonna want to check for `Array` here as well.